numba_cuda/numba/cuda/cuda_paths.py

```diff
@@ -3,14 +3,34 @@
 import os
 from collections import namedtuple
 import platform
-
+import site
+from pathlib import Path
 from numba.core.config import IS_WIN32
-from numba.misc.findlib import find_lib, find_file
+from numba.misc.findlib import find_lib
 from numba import config
+import glob
+import ctypes
 
 
 _env_path_tuple = namedtuple("_env_path_tuple", ["by", "info"])
 
+SEARCH_PRIORITY = [
+    "Conda environment",
+    "Conda environment (NVIDIA package)",
+    "NVIDIA NVCC Wheel",
+    "CUDA_HOME",
+    "System",
+    "Debian package",
+]
```

Comment:
What happens if someone has `LD_LIBRARY_PATH` set that points to something like `/usr/local/cuda/lib64`? That would end up bypassing whatever priority we have set here and make the underlying ctypes calls resolve there before resolving to any of these locations. We should probably account for that?

Reply:
Connected offline and when a resolved path contains a `/` character, it bypasses `LD_LIBRARY_PATH` in the underlying ctypes / `dlopen` calls so this isn't a problem. Resolving.